Classes/Controller/GalleryController.php

```diff
@@ -53,21 +53,76 @@ public function listAction($offset = 0) {
 			$imageItems = $this->fileCollectionService->getFileObjectsFromCollection($collectionUids);
 			$cObj = $this->configurationManager->getContentObject();
 			$currentUid = $cObj->data['uid'];
+			$columnPosition = $cObj->data['colPos']; //To determine where the gallery is placed (e.g. sidebar)
+
+			//If more than one FileCollection is selected
+			if (sizeof($collectionUids) > 1) {
+				//if a special gallery is requested
+				if ($this->request->hasArgument('galleryUID')) {
+					$gallery = array($this->request->getArgument('galleryUID'));
+					$imageItems = $this->fileCollectionService->getFileObjectsFromCollection($gallery);
+					$showBackToGallerySelectionLink = true;
+				} else {
+					$this->redirect('nested');
+				}
+			} else {
+				//If only one filecollection is selected
+				$imageItems = $this->fileCollectionService->getFileObjectsFromCollection($collectionUids);
+			}
 
 			$paginationArray = array(
 				'itemsPerPage' => $this->settings['imagesPerPage'],
 				'maximumVisiblePages' => $this->settings['numberOfPages'],
-				'insertAbove' => $this->settings['insertAbove'],
-				'insertBelow' => $this->settings['insertBelow']
+				//don't show pagination if there is only one page
+				'insertAbove' => (sizeof($imageItems) <= $this->settings['imagesPerPage']) ? false : $this->settings['insertAbove'],
```

Comment:
This could be configurable and set to true by default.
plugin.xyz.settings.hidePaginationIfOnlyOnePage or anything like that :)